docs/gc_overview.md

```diff
@@ -23,81 +23,183 @@
+A *mark* operation identifies which objects on the Java heap are reachable from outside of the heap and which objects are unreachable. Reachable objects are in use and must be retained, whereas unreachable objects are not in use and can be cleared away as *garbage*.
 
-A TLH is predefined with an initial default size of 2 KB. On every TLH refresh, the requested size for that thread
-is increased by an increment (default 4 KB). The requested size can grow up to a predefined maximum (default 128 KB).
+The process of marking involves a bit array called a *mark map* that is allocated by the VM at startup, based on the maximum heap size setting. Each bit in the *mark map* corresponds to 8 bytes of heap space. When an object is *marked*, its location in the heap is recorded by setting the appropriate bit in the *mark map*.
 
-After every GC cycle, the TLH requested size for each thread is reduced, sometimes by as much as 50%, to take account of threads that
-reduce their allocation rate and no longer need large TLHs.
+A *mark* operation can be broken down into the following phases:
 
-For very inactive threads, the requested size can even drop below the initial value, down to the predefined minimum (512/768 bytes).
-For very active threads, the maximum TLH requested size might be reached before the next GC occurs.
+1. Initial phase
 
-Larger TLHs can help reduce heap lock contention, but might also reduce heap utilisation and increase heap fragmentation.
+    A root object is an object that is accessible from outside of the heap such as a stack, class static field, or JNI reference. For other objects in the heap to be reachable, they must retain a connection from a root object. In the initial phase, tracing identifies all root objects by running a root scan. Root objects are pushed onto a work stack for processing in the next phase.
 
-The following options control the requested TLH size:
+2. Main phase
 
-- [`-Xgc:tlhMaximumSize=<bytes>`](xgc.md#tlhmaximumsize)
-- [`-Xgc:tlhInitialSize=<bytes>`](xgc.md#tlhinitialsize)
-- [`-Xgc:tlhIncrementSize=<bytes>`](xgc.md#tlhincrementsize)
+    The list of reachable root objects in the work stack is recursively traced for references to other objects in the heap. If new objects are found, they are added to the work stack. If an object is reachable, the appropriate bit is set in the *mark map*.
 
-Typically, when the maximum TLH size is increased, you should also increase the increment proportionally, so that active threads can
-reach the maximum requested TLH size more quickly.
+3. Final phase
 
-### SOA and LOA
+    The final phase processes weakly reachable roots such as finalizable objects, weak references, monitor sets, and string sets. If reachable objects are found, the appropriate bit is set in the *mark map*. For more information about the processing of *soft*, *weak*, and *phantom* references, see [Weak reference processing](#weak-reference-processing).
```

Comment:
While this is generally correct statement, this is not what defines this processing step:
If reachable objects are found, the appropriate bit is set in the *mark map*.

GC maintains special sets of root objects called weakly reachable roots such as finalizable objects, weak references..... Those roots do not determine liveness of objects. Those objects may or may not survive by virtue of having direct or indirect references from standard (strong) roots, as found by the main tracing phase.
If such object survives, nothing special is done. But if does not survive, it is removed from its weak root set, and optionally some special action is performed (such as reference enqueuing or execution of finalization method).

Reply:
OK, I removed that sentence in each operation that had it listed.